tests/sync/session.cpp

```diff
@@ -161,6 +165,38 @@ TEST_CASE("SyncSession: management by SyncUser", "[sync]") {
         EventLoop::main().run_until([&] { return session_is_active(*session1) && session_is_active(*session2); });
         REQUIRE(user->all_sessions().size() == 2);
     }
+
+    SECTION("a SyncUser properly allows registration of sessions that were recreated") {
+        const std::string user_id = "user1e";
+        const std::string path = "/test1e";
+        std::weak_ptr<SyncSession> weak_session;
+        std::string on_disk_path;
+        SyncConfig config;
+        auto user = SyncManager::shared().get_user(user_id, "not_a_real_token");
+        {
+            // Create the session within a nested scope, so we can control its lifetime.
+            std::string* ptr_to_path;
+            auto session = sync_session(server, user, path,
+                                        [&](auto...) { return s_test_token; },
+                                        [&](auto...) { },
+                                        SyncSessionStopPolicy::Immediately,
+                                        &ptr_to_path);
+            weak_session = session;
+            config = session->config();
+            REQUIRE(ptr_to_path);
+            on_disk_path = *ptr_to_path;
+            free(ptr_to_path);
+            REQUIRE(weak_session.lock() != nullptr);
+        }
+        // Session is dead, so the SyncUser's weak pointer to it should be nulled out.
+        REQUIRE(weak_session.lock() == nullptr);
+        // The next time we request it, it'll be created anew.
+        // The call to `get_session()` should result in `SyncUser::register_session()` being called.
+        auto session = SyncManager::shared().get_session(on_disk_path, config);
+        CHECK(session != nullptr);
```

Comment:
Is it necessary to explicitly compare against `nullptr`, or does `CHECK(session)` compile here?